### PR TITLE
feat: can use filter in graphql subqueries

### PR DESCRIPTION
**Pull request by mswertz**

### CODE REVIEW

**Review by connoratrug — CHANGES_REQUESTED** [review]
missing test and docs update

**Comment by jelmerveen** [comment]
When testing limit, offset, order, it all works. Except it will not throw an error when you try to order on a nested property that does not exist:

```
{
  Pet(filter: {name: {like: "e"}}) {
    name,
    tags(orderby: { order: DESC}, filter: {name: {like: "e"}}) {
      name
    }
  }
}
```

should result in error. Top level does this correctly:

```
{
  "error": {
    "errors": [
      {
        "message": "Error: Validation error of type WrongType: argument 'orderby' with value 'ObjectValue{objectFields=[ObjectField{name='order', value=EnumValue{name='DESC'}}]}' contains a field not in 'Petorderby': 'order' @ 'Pet'"
      }
    ]
  }
}
```

File: backend/molgenis-emx2/src/main/java/org/molgenis/emx2/SelectColumn.java [hunk 1]
```diff
@@ -9,10 +11,12 @@ public class SelectColumn {
   private Map<String, SelectColumn> children = new LinkedHashMap<>();
   private int limit = 0;
   private int offset = 0;
+  private Filter filter;
   private Map<String, Order> orderBy = new LinkedHashMap<>();
 
   public SelectColumn(String column) {
     this.column = column;
+    this.filter = new FilterBean(AND);
```
Comment (jelmerveen) [inline backend/molgenis-emx2/src/main/java/org/molgenis/emx2/SelectColumn.java:19]: So only and queries are possible within the nested filter?

Comment (mswertz) [reply backend/molgenis-emx2/src/main/java/org/molgenis/emx2/SelectColumn.java:19]: no. This is just the default.

**Comment by mswertz** [comment]
> maybe add the error when applying the orderby on the nested filter which does not exist

added, please verify

**Comment by jelmerveen** [comment]
Error is not showing in the playground?

**Comment by jelmerveen** [comment]
haha! it is showing, but not on 'order' specifically!

```
{
  Pet(orderby: { name: DESC}, filter: {name: {like: "e"}}) {
    name,
    tags(orderby: { t: DESC},filter: {name: {like: "e"}}) {
      name
    }
  }
}
```

gives error,

```
{
  Pet(orderby: { name: DESC}, filter: {name: {like: "e"}}) {
    name,
    tags(orderby: { order: DESC},filter: {name: {like: "e"}}) {
      name
    }
  }
}
```

gives no error

**Review by connoratrug — APPROVED** [review]

**Review by jelmerveen — APPROVED** [review]

**Pull request merged**